.pylintrc

```diff
@@ -107,7 +107,52 @@ ignore-imports=yes
 
 [MESSAGES CONTROL]
 
-disable=consider-using-ternary,locally-disabled,locally-enabled,logging-not-lazy,abstract-method,arguments-differ,broad-except,duplicate-code,fixme,inconsistent-return-statements,len-as-condition,missing-raises-doc,multiple-constructor-doc,no-else-return,no-member,no-self-use,not-context-manager,redefined-variable-type,redundant-returns-doc,too-many-arguments,too-many-boolean-expressions,too-many-branches,too-many-instance-attributes,too-many-lines,too-many-locals,too-many-public-methods,too-many-nested-blocks,too-many-statements,wrong-import-order
+disable=abstract-method,
+        arguments-differ,
+        broad-except,
+        consider-using-ternary,
+        duplicate-code,
+        fixme,
+        inconsistent-return-statements,
+        len-as-condition,
+        locally-disabled,
+        locally-enabled,
+        logging-not-lazy,
+        multiple-constructor-doc,
+        no-else-return,
+        no-member,
+        no-self-use,
+        not-context-manager,
+        redefined-variable-type,
+        redundant-returns-doc,
+        too-many-arguments,
+        too-many-boolean-expressions,
+        too-many-branches,
+        too-many-instance-attributes,
+        too-many-lines,
+        too-many-locals,
+        too-many-nested-blocks,
+        too-many-public-methods,
+        too-many-statements,
+        wrong-import-order,
+# TODO(#12912): Remove these after the Python 3 migration.
+        arg-name-for-non-keyword-arg,
+        arguments-renamed,
+        consider-using-dict-items,
+        consider-using-with,
+        cyclic-import,
+        deprecated-method,
+        import-outside-toplevel,
+        missing-raises-doc,
+        no-else-break,
+        no-else-continue,
+        no-else-raise,
+        non-explicit-keyword-args,
+        not-an-iterable,
+        raise-missing-from,
+        super-with-arguments,
+        unnecessary-pass,
+        useless-object-inheritance
```

Comment:
Also, instead of disabling these checks, can we use bypass flags in some of the places and add a TODO to remove those? (The current changes allows adding more unexpected patterns in the codebase)

Reply:
These are newly added checks in py3, when we were adding these we evaluated the severity and time required to removed those. Most of these are not that severe and would require around two or three people-days to fix, which is a lot and we want this PR to get in ASAP. I added the one common TODO for all of these and they can be fixed directly after we merge the PR.